packages/react-reconciler/src/ReactFiberScheduler.js

```diff
@@ -2111,6 +2121,82 @@ function warnIfNotCurrentlyActingUpdatesInDEV(fiber: Fiber): void {
 
 export const warnIfNotCurrentlyActingUpdatesInDev = warnIfNotCurrentlyActingUpdatesInDEV;
 
+let componentsWithSuspendedDiscreteUpdates = null;
+function checkForWrongSuspensePriorityInDEV(
+  thrownValue,
+  root,
+  sourceFiber,
+  expirationTime,
+) {
+  if (__DEV__) {
+    if (
+      // Check if the thrown value is a thenable
+      typeof thrownValue === 'object' &&
+      typeof thrownValue.then === 'function' &&
+      (sourceFiber.mode & ConcurrentMode) !== NoEffect &&
+      // Check if we're currently rendering a discrete update. Ideally, all we
+      // would need to do is check the current priority level. But we currently
+      // have no rigorous way to distinguish work that was scheduled at user-
+      // blocking priority from work that expired a bit and was "upgraded" to
+      // a higher priority. That's because we don't schedule separate callbacks
+      // for every level, only the highest priority level per root. The priority
+      // of subsequent levels is inferred from the expiration time, but this is
+      // an imprecise heuristic.
+      //
+      // However, we do store the last discrete pending update per root. So we
+      // can reliably compare to that one. (If we broaden this warning to include
+      // high pri updates that aren't discrete, then this won't be sufficient.)
```

Comment:
We really should have this warning too since it's pretty important to fix the mousemove events to have this property and we really shouldn't suspend there.

Reply:
We can get back to it later. I suspect I'll need to track some new field for suspendIfNeeded. Maybe we can use that.